### PR TITLE
removing deep copies from list_struct and modifier_change

### PR DESCRIPTION
**Pull request by vicdoval**
added some deepcopy=False to nodes of list_struct (they didn't need the deepcopy)

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

File: nodes/modifier_change/extrude_separate.py [hunk 1]
```diff
@@ -195,16 +195,13 @@ def process(self):
...
-        for vertices, edges, faces, masks, heights, scales, matrixes, face_data in zip(*meshes):
+        for vertices, edges, faces, masks_, heights_, scales_, matrixes_, face_data in zip(*meshes):
 
             new_extruded_faces = []
             new_extruded_faces_append = new_extruded_faces.append
-            fullList(heights, len(faces))
-            fullList(scales, len(faces))
-            fullList(matrixes, len(faces))
-            fullList(masks, len(faces))
+            heights, scales, matrixes, masks = make_repeaters([heights_, scales_, matrixes_, masks_])
```
Comment (portnov) [inline nodes/modifier_change/extrude_separate.py:202]: Shouldn't this depend on len(faces)?

**Comment by portnov** [comment]
Do I understand correctly that you've removed all usages of `fullList` from nodes?
If so, then I suggest marking the method with `@deprecate` (from sverchok.utils.decorators), so that people who use the method in scripts would see warnings. After some time this method can be removed.

File: nodes/modifier_change/extrude_separate.py (hunk 1 above)
Comment (vicdoval) [reply nodes/modifier_change/extrude_separate.py:202]: It creates iterators that later are used in zip(faces, repeater), so depends of faces but does not create the full list
```
def make_repeaters(lists):
    chain = itertools.chain
    repeat = itertools.repeat
    out =[]
    for l in lists:
        out.append(chain(l, repeat(l[-1])))

    return out
```

**Comment by vicdoval** [comment]
At this point (when I'm thinking to merge this PR) there are still 30 nodes using `fullList` function so maybe we are not ready to depreciate the function

**Comment by portnov** [comment]
ok

**Pull request merged**